webapp/src/components/backstage/playbook_runs/playbook_run_backstage/retrospective/retrospective.tsx

```diff
@@ -1,33 +1,77 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React from 'react';
-import {useIntl} from 'react-intl';
-
 import styled from 'styled-components';
 
+import React, {useEffect, useRef, useState} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import {DateTime} from 'luxon';
+
+import debounce from 'debounce';
+
 import UpgradeRetrospectiveSvg from 'src/components/assets/upgrade_retrospective_svg';
-import {Container, Left, Right} from 'src/components/backstage/playbook_runs/shared';
+import {Container, Content, Left, Right, Title} from 'src/components/backstage/playbook_runs/shared';
 import UpgradeBanner from 'src/components/upgrade_banner';
 import {AdminNotificationType} from 'src/constants';
 
-import {useAllowRetrospectiveAccess} from 'src/hooks';
-import {PlaybookRun} from 'src/types/playbook_run';
+import {useAllowPlaybookAndRunMetrics, useAllowRetrospectiveAccess} from 'src/hooks';
+import {PlaybookRun, RunMetricData} from 'src/types/playbook_run';
+import {Metric} from 'src/types/playbook';
+
+import {publishRetrospective, updateRetrospective} from 'src/client';
+
+import {PrimaryButton} from 'src/components/assets/buttons';
+
+import {Timestamp} from 'src/webapp_globals';
+
+import ConfirmModalLight from 'src/components/widgets/confirmation_modal_light';
+
+import MetricsData from '../metrics/metrics_data';
 
 import Report from './report';
+
 import TimelineRetro from './timeline_retro';
 
+const editDebounceDelayMilliseconds = 2000;
+
 interface Props {
     playbookRun: PlaybookRun;
+    metricsConfigs: Metric[] | null;
```

Comment:
I make a suggestion below in `playbook_run_backstage` about the `||`, that lets you do this:
```suggestion
    metricsConfigs?: Metric[];
```